src/mbgl/gl/index_buffer_resource.cpp

```diff
@@ -1,10 +1,18 @@
 #include <mbgl/gl/context.hpp>
 #include <mbgl/gl/index_buffer_resource.hpp>
+#include <mbgl/util/instrumentation.hpp>
 
 namespace mbgl {
 namespace gl {
 
+IndexBufferResource::IndexBufferResource(UniqueBuffer&& buffer_, int byteSize_)
+    : buffer(std::move(buffer_)),
+      byteSize(byteSize_) {
+    MLN_TRACE_ALLOC_INDEX_BUFFER(buffer.get(), byteSize);
```

Comment:
This location as well as others (vertex buffers, texture resources, render buffer creation) reinterpret_cast a `uint32_t` to a `const void*` which is a warning (C4312) in MSVC. The library builds with /WX / warnings as errors so this prevents MSVC from building with tracy enabled. Maybe try using the address of the `unique_resource` instead of the wrapped `BufferID`? Or disable warnings inside each macro for MSVC `#pragma warning(push), ...`